### PR TITLE
fix: create a method to get direct access to authorize response params

### PR DESCRIPTION
**Pull request by leleueri**
this help to manage the form_post response_mode

fixes AM-763

gravitee-io/issues#9179

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/response/AuthorizationResponse.java [hunk 1]
```diff
@@ -56,6 +57,7 @@ public void setRedirectUri(String redirectUri) {
     }
 
     public abstract String buildRedirectUri() throws URISyntaxException;
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/response/AuthorizationResponse.java:59]: 'java.net.URISyntaxException' is never thrown in the method

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/response/jwt/JWTAuthorizationResponse.java [hunk 2]
```diff
@@ -159,4 +160,11 @@ public String buildRedirectUri() throws URISyntaxException {
 
         return uriBuilder.buildString();
     }
+
+    @Override
+    public MultiMap params() {
+        MultiMap result = MultiMap.caseInsensitiveMultiMap();
+        result.add(io.gravitee.am.common.oidc.Parameters.RESPONSE, token);
```
Comment (mstasgravitee) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/response/jwt/JWTAuthorizationResponse.java:167]: Suggestion: 
there can be used a static import io.gravitee.am.common.oidc.Parameters.RESPONSE

File: gravitee-am-service/src/main/java/io/gravitee/am/service/utils/vertx/RequestUtils.java [hunk 3]
```diff
@@ -93,13 +93,17 @@ public static MultiMap getCleanedQueryParams(HttpServerRequest httpServerRequest
      */
     public static MultiMap getCleanedQueryParams(String uri) {
         final MultiMap queryParams = getQueryParams(uri);
-        queryParams.remove(ConstantKeys.ERROR_PARAM_KEY);
```
Comment (mstasgravitee) [inline gravitee-am-service/src/main/java/io/gravitee/am/service/utils/vertx/RequestUtils.java:96]: why are 96-102 lines removed? If it is by purpose the method's comment is invalid.

**Review by mstasgravitee — CHANGES_REQUESTED** [review]

File: gravitee-am-service/src/main/java/io/gravitee/am/service/utils/vertx/RequestUtils.java (hunk 3 above)
Comment (leleueri) [reply gravitee-am-service/src/main/java/io/gravitee/am/service/utils/vertx/RequestUtils.java:96]: > why are 96-102 lines removed? If it is by purpose the method's comment is invalid.

I do not get you point. I have extracted a method named cleanParams, behaviour should be the same

**Review by mstasgravitee — APPROVED** [review]

**Comment by leleueri** [comment]
Tests are OK on other PSQL version and both Mongo version. Merge the PR

**Pull request merged**